gvm/protocols/ospv1.py

```diff
@@ -18,14 +18,55 @@
 """
 Module for communication to a daemon speaking Open Scanner Protocol version 1
 """
-from gvm.protocols.base import GvmProtocol
+import logging
+
 from gvm.utils import get_version_string
 from gvm.xml import XmlCommand
 
+from .base import GvmProtocol
+
+logger = logging.getLogger(__name__)
+
 PROTOCOL_VERSION = (1, 2,)
 
+
+def create_credentials_element(_xmlcredentials, credentials):
+    """Generates an xml element with credentials."""
+    for service, credential in credentials.items():
+        cred_type = credential.get('type')
+        serv_port = credential.get('port')
+        username = credential.get('username')
+        password = credential.get('password')
+        _xmlcredential = _xmlcredentials.add_element('credential',
+                                                     attrs={'type': cred_type,
+                                                            'port': serv_port,
+                                                            'service': service,
+                                                     })
+        _xmlcredential.add_element('username', username)
+        _xmlcredential.add_element('password', password)
+    return _xmlcredentials
+
+def create_vt_selection_element(_xmlvtselection, vt_selection):
+    """Generates an xml element with a selection of Vulnerability tests."""
+    for vt_id, vt_values in vt_selection.items():
+        if vt_id != 'vt_groups':
+            _xmlvt = _xmlvtselection.add_element('vt_single',
+                                                 attrs={'id': vt_id})
+            if vt_values:
+                for key, value in vt_values.items():
+                    _xmlvt.add_element('vt_value', value, attrs={'id': key})
+        else:
+            for group in vt_values:
+                _xmlvt = _xmlvtselection.add_element('vt_group',
+                                                 attrs={'filter': group})
+
+    return _xmlvtselection
+
 class Osp(GvmProtocol):
 
+    def __init__(self, connection, transform=None):
+        super().__init__(connection, transform)
+
```

Comment:
Is the constructor required here? I am not sure anymore how the __init__ call sequence does work in python.

Reply:
It is not required. So, removed. Thanks!